src/v1/destinations/emarsys/networkHandler.js

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+const { isObject } = require('@rudderstack/integrations-lib');
+const { TransformerProxyError } = require('../../../v0/util/errorTypes');
+const { prepareProxyRequest, proxyRequest } = require('../../../adapters/network');
+const { isHttpStatusSuccess } = require('../../../v0/util/index');
+
+const {
+  processAxiosResponse,
+  getDynamicErrorType,
+} = require('../../../adapters/utils/networkUtils');
+const tags = require('../../../v0/util/tags');
+
+// ref : https://dev.emarsys.com/docs/emarsys-core-api-guides/c47a64a8ea7dc-http-200-errors
+function checkIfEventIsAbortableAndExtractErrorMessage(event, destinationResponse, keyId) {
+  const { errors } = destinationResponse.response.data;
+
+  // Determine if event is a string or an object, then fetch the corresponding key or value
+  let errorKey;
+  if (typeof event === 'string') {
+    errorKey = event;
+  } else if (typeof event === 'object' && event[keyId]) {
+    errorKey = event[keyId];
+  } else {
+    return { isAbortable: false, errorMsg: '' }; // Early return if neither condition is met or if keyId is missing in the object
+  }
+
+  // Check if this key has a corresponding error in the errors object
+  if (errors && isObject(errors) && errors[errorKey]) {
+    // const errorCode = Object.keys(errors[errorKey])[0]; // Assume there is at least one error code
+    const errorMsg = JSON.stringify(errors[errorKey]);
+    return { isAbortable: true, errorMsg };
+  }
+
+  // if '<NO_KEY_ID>' is present in the error object, that means, it is a root level error, and none of the events are supposed to be successful
+  if (errors && isObject(errors) && errors['<NO_KEY_ID>']) {
+    const errorMsg = JSON.stringify(errors['<NO_KEY_ID>']);
+    return { isAbortable: true, errorMsg };
+  }
+
+  // Return false and an empty error message if no error is found
+  return { isAbortable: false, errorMsg: '' };
+}
+
+const responseHandler = (responseParams) => {
+  const { destinationResponse, rudderJobMetadata, destinationRequest } = responseParams;
```

Comment:
@shrouti1507 destType is 3rd parameter, can you recheck the contract updated in native integration.ts ?

Reply:
<img width="1363" alt="Screenshot 2024-05-14 at 3 03 47 PM" src="https://github.com/rudderlabs/rudder-transformer/assets/60211312/3492fe04-6b76-4265-93c6-2e5fe855c283">
@krishna2020 added it here

Reply:
see the contract in nativeIntegration.ts & in networkHandler.js
you are passing destType & referring it as destinationRequest here

Reply:
did not get you, we are passing response params, which is an object, and in that object we have introduced a key called destinatination Request. And we are destructuring it there. Just for a simple example like below 
<img width="881" alt="Screenshot 2024-05-14 at 4 06 59 PM" src="https://github.com/rudderlabs/rudder-transformer/assets/60211312/f8045bb7-0644-452d-836e-030c67d2db97">